test/integration/join_cluster_test.js

```diff
@@ -0,0 +1,215 @@
+    assert.plan(4);
+
+    var done = after(2, function onDone() {
+        destroyCluster(cluster);
+        assert.end();
+    });
+
+    function onBootstrap(err, nodesJoined) {
+        assert.ifError(err, 'no bootstrap error');
+        assert.equal(nodesJoined.length, 1, 'joined one other node');
+        done();
+    }
+
+    var cluster = bootstrapClusterOf(2, null, null, onBootstrap);
+});
+
+test('three nodes can join', function t(assert) {
+    assert.plan(6);
+
+    var done = after(3, function onDone() {
+        destroyCluster(cluster);
+        assert.end();
+    });
+
+    function onBootstrap(err, nodesJoined) {
+        assert.ifError(err, 'no bootstrap error');
+        assert.equal(nodesJoined.length, 2, 'joined one other node');
+        done();
+    }
+
+    var cluster = bootstrapClusterOf(3, null, null, onBootstrap);
+});
+
+test('three nodes, 1 bad node, join size 1', function t(assert) {
+    var done = after(3, function onDone() {
+        destroyCluster(cluster);
+        assert.end();
+    });
+
+    function onBootstrap(err, nodesJoined) {
+        assert.ifError(err, 'no bootstrap error');
+        assert.ok(nodesJoined.length >= 1, 'joined at least one other node');
+        assert.ok(nodesJoined.indexOf(cluster[2].hostPort) === -1, 'no one can join this guy');
+        done();
+    }
+
+    var cluster = bootstrapClusterOf(3, {
+        joinSize: 1
+    }, function tap(cluster) {
+        cluster[2].denyJoins();
+    }, onBootstrap);
+});
+
+test('three nodes, 1 bad node, join size 2', function t(assert) {
+    var done = after(3, function onDone() {
+        destroyCluster(cluster);
+        assert.end();
+    });
+
+    function onBootstrap(err) {
+        assert.ok(err, 'bootstrap error occurred');
+        done();
+    }
+
+    var cluster = bootstrapClusterOf(3, {
+        joinSize: 2,
+        maxJoinDuration: 100
+    }, function tap(cluster) {
+        cluster[1].denyJoins();
+        cluster[2].denyJoins();
+    }, onBootstrap);
+});
+
+test('mega cluster', function t(assert) {
+    var done = after(25, function onDone() {
+        destroyCluster(cluster);
+        assert.end();
+    });
+
+    function onBootstrap(err) {
+        assert.ifError(err, 'no bootstrap error occurred');
+        done();
+    }
+
+    var cluster = bootstrapClusterOf(25, null, null, onBootstrap);
+});
+
+test('slow joiner', function t(assert) {
+    var done = after(2, function onDone() {
+        assert.end();
+    });
+
+    function onFastBootstrap(err) {
+        assert.ifErr(err, 'no bootstrap error');
+        done();
+    }
+
+    function onSlowBootstrap(err) {
+        assert.equal(err.type, 'ringpop.join-aborted', 'join aborted error');
+        done();
+    }
+
+    bootstrapClusterOf(2, null, function tap(cluster) {
+        cluster[1].protocolJoin = function protocolJoin(options, callback) {
+            setTimeout(function onTimeout() {
+                destroyCluster(cluster);
```

Comment:
destroying here causes the abort. it's part of the test. ill put a comment. since this is non-obvious, ill put a comment.